xtask/src/starfive/visionfive2.rs

```diff
@@ -1,266 +1,126 @@
-
-    let mut hdr = vec![];
-    hdr.extend_from_slice(&spl_header_offset);
-    hdr.extend_from_slice(&default_backup); // Offset of backup SBL from Flash info start (from input_sbl_normal.cfg)
-    hdr.resize(hdr.len() + 636, 0);
-    hdr.extend_from_slice(&default_version_id);
-
-    let l: [u8; 4] = unsafe { transmute({ dat.len() as u32 }.to_le()) };
-    hdr.extend_from_slice(&l); /* u-boot-spl.bin size in bytes */
-    hdr.extend_from_slice(&data_offset);
-    let l: [u8; 4] = unsafe { transmute(fv.to_le()) };
-    hdr.extend_from_slice(&l); /* CRC32 of u-boot-spl.bin */
-    hdr.resize(hdr.len() + 364, 0);
-    assert!(hdr.len() == 1024, "hdr is {:?} bytes, not 1024", hdr.len());
-    hdr.extend(&dat);
-    hdr
-}
+const ARCH: &str = "riscv64";
+const TARGET: &str = "riscv64imac-unknown-none-elf";
 
-#[test]
-fn test_hdr() {
-    static HOSTS: &'static [u8] = include_bytes!("testdata/hosts");
-    static HOSTS_OUT: &'static [u8] = include_bytes!("testdata/hosts.normal.out");
+const BT0_BIN: &str = "starfive-visionfive2-bt0.bin";
+const BT0_ELF: &str = "starfive-visionfive2-bt0";
 
-    let out = spl_create_hdr(HOSTS.to_vec());
-    assert_eq!(HOSTS_OUT.len(), out.len());
-    for (x, val) in out.iter().enumerate() {
-        if *val != HOSTS_OUT[x] {
-            println!("at index {x:#x} got {:#x}, want {:#x}", *val, HOSTS_OUT[x]);
-        }
-    }
-    assert_eq!(HOSTS_OUT, &out);
-}
+const MAIN_BIN: &str = "starfive-visionfive2-main.bin";
+const MAIN_ELF: &str = "starfive-visionfive2-main";
 
-// CRC-32-IEEE being the most commonly used one
+const BOARD_DTB: &str = "starfive-visionfive2-board.dtb";
 
-// const SRAM0_SIZE = 128 * 1024;
-const SRAM0_SIZE: u64 = 32 * 1024;
+// TODO: find great names
+const FDT_BIN: &str = "starfive-visionfive2-board.fdtbin";
+const SRAM_IMAGE: &str = "starfive-visionfive2-sram.bin";
 
-const DEFAULT_TARGET: &str = "riscv64imac-unknown-none-elf";
+const IMAGE_BIN: &str = "starfive-visionfive2.bin";
 
-pub(crate) fn execute_command(args: &crate::Cli, features: Vec<String>) {
+pub(crate) fn execute_command(args: &Cli, features: Vec<String>) {
     match args.command {
         Commands::Make => {
             info!("building VisionFive2");
-            let binutils_prefix = find_binutils_prefix_or_fail();
-            // bt0 stage
-            xtask_build_jh7100_flash_bt0(&args.env, &features);
-            xtask_binary_jh7100_flash_bt0(binutils_prefix, &args.env);
-            xtask_add_bt0_header(&args.env);
-            // main stage
-            xtask_build_jh7100_flash_main(&args.env);
-            xtask_binary_jh7100_flash_main(binutils_prefix, &args.env);
-            xtask_concat_flash_binaries(&args.env);
+            // Get binutils first so we can fail early
+            let binutils_prefix = &find_binutils_prefix_or_fail(ARCH);
+            // Build the stages - should we parallelize this?
+            xtask_build_jh7110_main(&args.env);
+            xtask_build_jh7110_bt0(&args.env, &features);
```

Comment:
This may not matter, but is there a reason for switching the build order of `main` and `bt0`?

Reply:
Yes, it should not matter here. It is a relic of iteration of approaches. I'll swap it back.

Reply:
done